lib/fs.js

```diff
@@ -255,11 +255,15 @@ Object.defineProperties(fs, {
 });
 
 function handleError(val, callback) {
+  if (typeof val === 'string')
```

Comment:
It's just optimizing for the case where `URL` instances are passed to `fs` methods. `handleError()` is passed the result of a function which converts `URL` instances to string paths. By checking for string path results first, we can avoid the `instanceof`, which is probably a less common case.